pkg/tools/clockmock/mock.go

```diff
@@ -42,20 +42,48 @@ func New() *Mock {
 	}
 }
 
+// Start starts mock time to run with the given speed until ctx becomes done. While time is running, current time for
+// the mock will be the following:
+//   mock time := mock start time  +  (real time duration from the start) * speed  +  mock duration added with Set, Add
+func (m *Mock) Start(ctx context.Context, speed float64) {
```

Comment:
1. Do we need stop method?
2. What if Start has called twice?

Reply:
1. Nope, because we have ctx.
2. Time will run with (speed-1 + speed-2) speed :)

Reply:
Should this operation be thread-blockable?

Reply:
Reworked with always running time and `Mock.SetSpeed`.